src/types.rs

```diff
@@ -152,7 +152,7 @@ impl Readable for UserChannelId {
 /// Details of a channel as returned by [`Node::list_channels`].
 ///
 /// [`Node::list_channels`]: crate::Node::list_channels
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone)]
```

Comment:
this is because 
`binary operation == cannot be applied to type std::sync::Arc<types::ChannelConfig>`